### PR TITLE
add optimizer factory

### PR DESCRIPTION
**Pull request by TomorrowIsAnOtherDay**

### CODE REVIEW

File: deepes/include/torch/es_agent.h [hunk 1]
```diff
@@ -17,7 +17,8 @@
 
 #include <memory>
 #include <string>
-#include "optimizer.h"
+#include <algorithm>
+#include "optimizer_factory.h"
```
Comment (rical730) [inline deepes/include/torch/es_agent.h:20]: missing  file optimizer_factory.h

Comment (TomorrowIsAnOtherDay) [reply deepes/include/torch/es_agent.h:20]: done

File: deepes/include/torch/es_agent.h [hunk 2]
```diff
@@ -17,7 +17,8 @@
 
 #include <memory>
 #include <string>
-#include "optimizer.h"
+#include <algorithm>
```
Comment (rical730) [inline deepes/include/torch/es_agent.h]: `include <algorithm>` can be deleted

Comment (TomorrowIsAnOtherDay) [reply deepes/include/torch/es_agent.h]: done

**Review by rical730 — APPROVED** [review]
Looks good to me.

**Pull request merged**